From 2fccda76e936e9a71d312b578a06c1b4eaec72d4 Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@denx.de>
Date: Thu, 7 May 2020 15:41:22 +0200
Subject: [PATCH 13/13] ARM: dts: stm32: Add GPIO LEDs for STM32MP1 DHCOM PDK2

Add bindings for the four GPIO LEDs on DH PDK2 board. Note that LED5
GPIO-E may conflict with touchscreen interrupt, hence LED5 must be
disabled when using the DH 560-200 display unit with touchscreen.

Signed-off-by: Marek Vasut <marex@denx.de>
---
 ...2mp157c-dhcom-pdk2-overlay-560-200-x12.dts | 11 ++++++++
 arch/arm/boot/dts/stm32mp157c-dhcom-pdk2.dts  | 28 +++++++++++++++++++
 2 files changed, 39 insertions(+)

diff --git a/arch/arm/boot/dts/stm32mp157c-dhcom-pdk2-overlay-560-200-x12.dts b/arch/arm/boot/dts/stm32mp157c-dhcom-pdk2-overlay-560-200-x12.dts
index 92d6922eecb9..1d6a30190904 100644
--- a/arch/arm/boot/dts/stm32mp157c-dhcom-pdk2-overlay-560-200-x12.dts
+++ b/arch/arm/boot/dts/stm32mp157c-dhcom-pdk2-overlay-560-200-x12.dts
@@ -77,6 +77,16 @@
 	};
 
 	fragment-3 {
+		target-path = "/led";
+		#address-cells = <1>;
+		#size-cells = <1>;
+		__overlay__ {
+			/* LED5 GPIO-E conflicts with touchscreen IRQ */
+			/delete-node/ led-0;
+		};
+	};
+
+	fragment-4 {
 		target-path = "/soc/i2c@40015000";
 		#address-cells = <1>;
 		#size-cells = <1>;
@@ -86,6 +96,7 @@
 			polytouch@38 {
 				compatible = "edt,edt-ft5x06";
 				reg = <0x38>;
+				/* Touchscreen IRQ GPIO-E conflicts with LED5 GPIO */
 				interrupt-parent = <&gpiog>;
 				interrupts = <2 IRQ_TYPE_EDGE_FALLING>; /* GPIO E */
 				linux,wakeup;
diff --git a/arch/arm/boot/dts/stm32mp157c-dhcom-pdk2.dts b/arch/arm/boot/dts/stm32mp157c-dhcom-pdk2.dts
index 2a3398227666..e4a0a8b5c2d6 100644
--- a/arch/arm/boot/dts/stm32mp157c-dhcom-pdk2.dts
+++ b/arch/arm/boot/dts/stm32mp157c-dhcom-pdk2.dts
@@ -81,6 +81,34 @@
 		};
 	};
 
+	led {
+		compatible = "gpio-leds";
+
+		led-0 {
+			label = "green:led5";
+			gpios = <&gpiog 2 GPIO_ACTIVE_HIGH>;
+			default-state = "off";
+		};
+
+		led-1 {
+			label = "green:led6";
+			gpios = <&gpiod 11 GPIO_ACTIVE_HIGH>;
+			default-state = "off";
+		};
+
+		led-2 {
+			label = "green:led7";
+			gpios = <&gpioi 2 GPIO_ACTIVE_HIGH>;
+			default-state = "off";
+		};
+
+		led-3 {
+			label = "green:led8";
+			gpios = <&gpioi 3 GPIO_ACTIVE_HIGH>;
+			default-state = "off";
+		};
+	};
+
 	sound {
 		compatible = "audio-graph-card";
 		routing =
-- 
2.25.1

